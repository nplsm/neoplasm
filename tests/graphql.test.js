const fetch = require("node-fetch")

test("Correct fetch of items", async () => {
  const slug = "lifeoxetine"
  const response = await fetch("http://localhost:3001/", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      query: `
        query($slug: String!) {
          item(slug: $slug) {
            release {
              title
              cover
              tracks {
                _id
                position
                title
                artists {
                  alias
                }
                featuring {
                  alias
                }
                sources {
                  src
                  type
                }
              }
            }
            copies {
              _id
              shiped
            }
          }
        }
      `,
      variables: { slug },
    }),
  })
  const { data } = await response.json()
  const { item } = await data
  expect(item).toMatchSnapshot()
})
