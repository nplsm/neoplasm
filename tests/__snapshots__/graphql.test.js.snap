// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct fetch of items 1`] = `
Object {
  "copies": Array [
    Object {
      "_id": "4209348102a09234f80293b42b908903",
      "shiped": true,
    },
    Object {
      "_id": "4209234532047234f80293b42b908903",
      "shiped": false,
    },
  ],
  "release": Object {
    "cover": "https://cdn.neoplasm.xyz/images/covers/lifeoxetine.jpg",
    "title": "Lifeoxetine",
    "tracks": Array [
      Object {
        "_id": "601fb3db240815002823e6a4",
        "artists": Array [
          Object {
            "alias": "Marble Bust",
          },
        ],
        "featuring": Array [],
        "position": 1,
        "sources": Array [
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/mp3/01%20Marble%20Bust%20%E2%80%93%C2%A0THENEWERA.mp3",
            "type": "mp3",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/ogg/01%20Marble%20Bust%20%E2%80%93%C2%A0THENEWERA.ogg",
            "type": "ogg",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/wav/01%20Marble%20Bust%20%E2%80%93%C2%A0THENEWERA.wav",
            "type": "wav",
          },
        ],
        "title": "THENEWERA",
      },
      Object {
        "_id": "601fb3db240815002823e6a9",
        "artists": Array [
          Object {
            "alias": "Marble Bust",
          },
        ],
        "featuring": Array [
          Object {
            "alias": "увула",
          },
        ],
        "position": 2,
        "sources": Array [
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/mp3/02%20Marble%20Bust%20%E2%80%93%C2%A0TRANSGRESSION%20%28feat.%20%D1%83%D0%B2%D1%83%D0%BB%D0%B0%29.mp3",
            "type": "mp3",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/ogg/02%20Marble%20Bust%20%E2%80%93%C2%A0TRANSGRESSION%20%28feat.%20%D1%83%D0%B2%D1%83%D0%BB%D0%B0%29.ogg",
            "type": "ogg",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/wav/02%20Marble%20Bust%20%E2%80%93%C2%A0TRANSGRESSION%20%28feat.%20%D1%83%D0%B2%D1%83%D0%BB%D0%B0%29.wav",
            "type": "wav",
          },
        ],
        "title": "TRANSGRESSION",
      },
      Object {
        "_id": "601fb3db240815002823e6af",
        "artists": Array [
          Object {
            "alias": "Marble Bust",
          },
        ],
        "featuring": Array [],
        "position": 3,
        "sources": Array [
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/mp3/03%20Marble%20Bust%20%E2%80%93%C2%A0ANGELS.mp3",
            "type": "mp3",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/ogg/03%20Marble%20Bust%20%E2%80%93%C2%A0ANGELS.ogg",
            "type": "ogg",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/wav/03%20Marble%20Bust%20%E2%80%93%C2%A0ANGELS.wav",
            "type": "wav",
          },
        ],
        "title": "ANGELS",
      },
      Object {
        "_id": "601fb3db240815002823e6ba",
        "artists": Array [
          Object {
            "alias": "Marble Bust",
          },
        ],
        "featuring": Array [],
        "position": 5,
        "sources": Array [
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/mp3/05%20Marble%20Bust%20%E2%80%93%C2%A0SORRYAU.mp3",
            "type": "mp3",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/ogg/05%20Marble%20Bust%20%E2%80%93%C2%A0SORRYAU.ogg",
            "type": "ogg",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/wav/05%20Marble%20Bust%20%E2%80%93%C2%A0SORRYAU.wav",
            "type": "wav",
          },
        ],
        "title": "SORRYAU",
      },
      Object {
        "_id": "601fb3db240815002823e6b4",
        "artists": Array [
          Object {
            "alias": "Marble Bust",
          },
        ],
        "featuring": Array [
          Object {
            "alias": "sparrow",
          },
        ],
        "position": 4,
        "sources": Array [
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/mp3/04%20Marble%20Bust%20%E2%80%93%C2%A0HINDSIGHT%20%28feat.%20SPARROW%29.mp3",
            "type": "mp3",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/ogg/04%20Marble%20Bust%20%E2%80%93%C2%A0HINDSIGHT%20%28feat.%20SPARROW%29.ogg",
            "type": "ogg",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/wav/04%20Marble%20Bust%20%E2%80%93%C2%A0HINDSIGHT%20%28feat.%20SPARROW%29.wav",
            "type": "wav",
          },
        ],
        "title": "HINDSIGHT",
      },
      Object {
        "_id": "601fb3db240815002823e6bf",
        "artists": Array [
          Object {
            "alias": "Marble Bust",
          },
        ],
        "featuring": Array [],
        "position": 6,
        "sources": Array [
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/mp3/06%20Marble%20Bust%20%E2%80%93%C2%A0TRANCEGRESSION.mp3",
            "type": "mp3",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/ogg/06%20Marble%20Bust%20%E2%80%93%C2%A0TRANCEGRESSION.ogg",
            "type": "ogg",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/wav/06%20Marble%20Bust%20%E2%80%93%C2%A0TRANCEGRESSION.wav",
            "type": "wav",
          },
        ],
        "title": "TRANCEGRESSION",
      },
      Object {
        "_id": "601fb3db240815002823e6c4",
        "artists": Array [
          Object {
            "alias": "Marble Bust",
          },
        ],
        "featuring": Array [],
        "position": 7,
        "sources": Array [
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/mp3/07%20Marble%20Bust%20%E2%80%93%C2%A0THELIFTINGHAND.mp3",
            "type": "mp3",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/ogg/07%20Marble%20Bust%20%E2%80%93%C2%A0THELIFTINGHAND.ogg",
            "type": "ogg",
          },
          Object {
            "src": "https://cdn.neoplasm.xyz/music/lifeoxetine/wav/07%20Marble%20Bust%20%E2%80%93%C2%A0THELIFTINGHAND.wav",
            "type": "wav",
          },
        ],
        "title": "THELIFTINGHAND",
      },
    ],
  },
}
`;
